src/twister2/builder/build_filter_processor.py

```diff
@@ -0,0 +1,36 @@
+from __future__ import annotations
+
+import logging
+
+from twister2.builder.builder_abstract import BuildConfig, BuilderAbstract
+from twister2.cmake_filter.cmake_filter import CMakeFilter
+from twister2.exceptions import TwisterBuildFiltrationException
+
+logger = logging.getLogger(__name__)
+
+
+class BuildFilterProcessor:
+    """Run additional setup before actual build is triggered."""
+
+    def __init__(self, builder: BuilderAbstract) -> None:
+        self.builder = builder
+
+    def process(self) -> None:
+        """Run build with cmake_helper flag enabled."""
+        self.builder.run_cmake_stage(True)
+        self.apply_cmake_filtration(self.builder.build_config)
+
+    @staticmethod
+    def apply_cmake_filtration(build_config: BuildConfig) -> None:
```

Comment:
Only `apply_cmake_filtration` is used now, so `process` method (and init with relation to `builder`) can be removed. However, it maybe useful if someone want to build own `builder_manager` fixture

Reply:
Yes, I decided to leave there whole this class and `process` method, because I expect that in the future we will use CMake package helper, and it can be done by this process method.